apps/angular/59-content-projection-defer/src/app/page-2.ts

```diff
@@ -36,8 +38,19 @@ interface Post {
   imports: [ExpandableCard],
 })
 export class Page2 {
-  public postRessource = httpResource<Post[]>(
-    'https://jsonplaceholder.typicode.com/posts',
-  );
   protected readonly ResourceStatus = ResourceStatus;
+  private readonly URL = 'https://jsonplaceholder.typicode.com/posts';
+
+  private readonly card = viewChild.required(ExpandableCard);
+  private readonly isCardOpened = computed(() => this.card().isExpanded());
+
+  protected readonly postResource = resource<Post[], { isCardOpened: boolean }>(
+    {
+      loader: async ({ request }) =>
+        request.isCardOpened
+          ? await fetch(this.URL).then((res) => res.json())
+          : [],
```

Comment:
i think how angular designed the API is to return undefined when you don't want to trigger the API call but nice solution.